cluster-operator/src/main/java/io/strimzi/operator/cluster/model/dynamicconfig/KafkaBrokerConfigurationBuilder.java

```diff
@@ -0,0 +1,360 @@
+    }
+
+    public KafkaBrokerConfigurationBuilder withZookeeper()  {
+        printSectionHeader("Zookeeper");
+        writer.println("zookeeper.connect=localhost:2181");
+        writer.println();
+
+        return this;
+    }
+
+    public KafkaBrokerConfigurationBuilder withListeners(String clusterName, String namespace, KafkaListeners kafkaListeners)  {
+        List<String> listeners = new ArrayList<>();
+        List<String> advertisedListeners = new ArrayList<>();
+        List<String> securityProtocol = new ArrayList<>();
+
+        // Replication listener
+        listeners.add("REPLICATION-9091://0.0.0.0:9091");
+        advertisedListeners.add(String.format("REPLICATION-9091://%s-${STRIMZI_BROKER_ID}.%s-brokers.%s.svc:9091", KafkaResources.kafkaStatefulSetName(clusterName), KafkaResources.kafkaStatefulSetName(clusterName), namespace));
+        securityProtocol.add("REPLICATION-9091:SSL");
+
+        printSectionHeader("Replication listener");
+        writer.println("listener.name.replication-9091.ssl.keystore.location=/tmp/kafka/cluster.keystore.p12");
+        writer.println("listener.name.replication-9091.ssl.keystore.password=${CERTS_STORE_PASSWORD}");
+        writer.println("listener.name.replication-9091.ssl.keystore.type=PKCS12");
+        writer.println("listener.name.replication-9091.ssl.truststore.location=/tmp/kafka/cluster.truststore.p12");
+        writer.println("listener.name.replication-9091.ssl.truststore.password=${CERTS_STORE_PASSWORD}");
+        writer.println("listener.name.replication-9091.ssl.truststore.type=PKCS12");
+        writer.println("listener.name.replication-9091.ssl.client.auth=required");
+        writer.println();
+
+        if (kafkaListeners != null) {
+            // PLAIN listener
+            if (kafkaListeners.getPlain() != null) {
+                printSectionHeader("Plain listener");
+
+                String listenerName = "PLAIN-9092";
+                listeners.add(listenerName + "://0.0.0.0:9092");
+                advertisedListeners.add(String.format("%s://%s-${STRIMZI_BROKER_ID}.%s-brokers.%s.svc:9092", listenerName, KafkaResources.kafkaStatefulSetName(clusterName), KafkaResources.kafkaStatefulSetName(clusterName), namespace));
+                configureAuthentication(writer, listenerName, securityProtocol, false, kafkaListeners.getPlain().getAuth());
+
+                writer.println();
+            }
+
+            if (kafkaListeners.getTls() != null) {
+                printSectionHeader("TLS listener");
+
+                String listenerName = "TLS-9093";
+                listeners.add(listenerName + "://0.0.0.0:9093");
+                advertisedListeners.add(String.format("%s://%s-${STRIMZI_BROKER_ID}.%s-brokers.%s.svc:9093", listenerName, KafkaResources.kafkaStatefulSetName(clusterName), KafkaResources.kafkaStatefulSetName(clusterName), namespace));
+                configureAuthentication(writer, listenerName, securityProtocol, true, kafkaListeners.getTls().getAuth());
```

Comment:
If I factor it out it would be a method with around 8 parameters for 3 lines of code. That does not seem to be worth it. I can probably reduce it a bit by changing the lists to class variables and not local variables. But it will be still a lot of parameters. If you think it is worth it, I can do it, but it doesn't seem to be helpful.

I factored out the advertised address part which had at least something in common.